app/policies/organizations/liked_pet_policy.rb

```diff
@@ -0,0 +1,27 @@
+class Organizations::LikedPetPolicy < ApplicationPolicy
+  authorize :pet, optional: true
+  pre_check :verify_adopter_foster_account!
+  pre_check :verify_pet_likable!, only: %i[create? destroy?]
+
+  def index?
+    permission?(:manage_liked_pets)
+  end
+
+  def create?
+    permission?(:manage_liked_pets)
+  end
+
+  def destroy?
+    permission?(:manage_liked_pets)
+  end
+
+  private
+
+  def verify_pet_likable!
+    deny! if pet.organization_id != user.organization_id
+  end
+
+  def verify_adopter_foster_account!
+    deny! unless user.adopter_foster_account.present?
+  end
+end
```

Comment:
I have left this pretty open ended such that an adopter or fosterer, even without a profile, can like and unlike any pet. I say we take this approach to start and let feedback inform us if we need to make changes. My assumption here is that people would like to be able to track liked pets before investing the time to fill out a profile. I see it as a way for users to shortlist pets before making the investment.